ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -800,14 +835,13 @@ runThreadNetwork ThreadNetworkArgs
       -- TODO Is there a risk that this will block because the 'forkTxProducer'
       -- fills up the mempool too quickly?
       forkCrucialTxs
+        clock
+        joinSlot
         registry
         -- a fingerprint for the ledger
-        ( (Origin, GenesisPoint)
-        , do
-            -- time matters, because some transaction expire
-            now <- WrappedClock.getCurrentSlot clock
-            p <- (ledgerTipPoint' (Proxy @blk) . ledgerState) <$> ChainDB.getCurrentLedger chainDB
-            pure (At now, p)
+        ( GenesisPoint
```

Comment:
The new argument `joinSlot` supplants the `WithOrigin Slot` component of the STM transaction's output.